src/lib/apiErrorHandler.js

```diff
@@ -144,3 +121,113 @@ export default async function handleErrors(req, res, error, logContext) {
     handleUnexpectedErrorInCatchBlock(req, res, e, logContext);
   }
 }
+
+/**
+ * Logs a worker error and stores it in the database.
+ * @param {Object} job - The job object.
+ * @param {string} operation - The operation name.
+ * @param {Error} error - The error object.
+ * @param {Object} logContext - The logging context.
+ * @returns {Promise<number|null>} - The ID of the stored request error, or null if storing failed.
+ */
+const logWorkerError = async (job, operation, error, logContext) => {
+  if (
+    operation !== 'SequelizeError'
+    && process.env.SUPPRESS_ERROR_LOGGING
+    && process.env.SUPPRESS_ERROR_LOGGING.toLowerCase() === 'true'
+  ) {
+    return 0;
+  }
+
+  try {
+    const responseBody = typeof error === 'object' && error !== null
+      ? { ...error, errorStack: error?.stack }
+      : error;
+
+    const requestBody = {
+      ...(job.data && typeof job.data === 'object' && Object.keys(job.data).length > 0 && { data: job.data }),
+    };
+
+    const requestErrorId = await createRequestError({
+      operation,
+      uri: job.queue.name,
+      method: 'PROCESS_JOB',
+      requestBody,
+      responseBody,
+      responseCode: INTERNAL_SERVER_ERROR,
+    });
+
+    return requestErrorId;
+  } catch (e) {
+    logger.error(`${logContext.namespace} - Sequelize error - unable to store RequestError - ${e}`);
+  }
+
+  return null;
+};
+
+/**
+ * Handles errors in a worker job.
+ * @param {Object} job - The job object.
+ * @param {Error} error - The error object.
+ * @param {Object} logContext - The context for logging.
+ */
+export const handleWorkerError = async (job, error, logContext) => {
+  if (process.env.NODE_ENV === 'development') {
+    logger.error(error);
+  }
+
+  let operation;
+  let label;
+
+  if (error instanceof Sequelize.Error) {
```

Comment:
why do we check instanceof Sequelize.Error here but above we do operation !== 'SequelizeError'.

Reply:
just matching original logic